### ISSUE TITLE
GPUs memory are not averge

### ISSUE THREAD

**Issue by alicera** [pytorch/pytorch#50251]
OS: ubuntu18.04
nvidia driver : 450.66
pytorch: '1.6.0a0+9907a3e'

![image](https://user-images.githubusercontent.com/59388989/103968556-062a0880-519f-11eb-84cb-8fa50c986f1c.png)

And I train the https://github.com/biubug6/Pytorch_Retinaface with 3GPUs.
3 GPUs and batch 15
The memory doesnt average.

**Comment by cyy5358** [comment]
Can torch.nn.parallel.DistributedDataParallel work better?
```python
    net = torch.nn.DataParallel(net).cuda()
```

**Comment by gchanan** [comment]
Please file an issue with Retinaface.

**Comment by alicera** [comment]
The code have this.  
  net = torch.nn.DataParallel(net).cuda()

I use the NGC pytorch 20.03,it can work.
But I use the pytorch 20.12, it will happen the error. If I set 4 GPU and 16 batch, it will happen one GPU get the 8 batch.

RuntimeError: CUDA out of memory. Tried to allocate 8.26 GiB (GPU 2; 10.92 GiB total capacity; 2.09 GiB already allocated; 8.26 GiB free; 2.14 GiB reserved in total by PyTorch)
